internal/services/storage/client/client.go

```diff
@@ -34,7 +34,7 @@ type Client struct {
 	BlobServicesClient *storage.BlobServicesClient
 	FileServicesClient *storage.FileServicesClient
 
-	authorizerForAad auth.Authorizer
+	authConfig *auth.Credentials
```

Comment:
I think it'd be worth calling this out as AzureAD/Entra specific credentials, since there's different authentication mechanisms here and `auth.Credentials` is unspecific?

```suggestion
	azureADAuthConfig *auth.Credentials
```